eaf.el

```diff
@@ -1677,15 +1704,50 @@ It currently identifies PDF, videos, images, and mindmap file extensions."
       (apply orig-fn file args))))
 (advice-add #'find-file :around #'eaf--find-file-advisor)
 
+(defcustom eaf-byte-compile-apps t "") 
+
 ;;;###autoload
-(defun eaf-install-and-update ()
+(defun eaf-install-and-update (&rest apps)
   "Interactively run `install-eaf.py' to install/update EAF apps.
 
 For a full `install-eaf.py' experience, refer to `--help' and run in a terminal."
   (interactive)
-  (let* ((eaf-dir (file-name-directory (locate-library "eaf")))
-         (default-directory eaf-dir))
-    (shell-command (concat eaf-python-command " install-eaf.py" "&"))))
+  (let* ((default-directory eaf-source-dir)
+	 (output-buffer (generate-new-buffer "*EAF installation*"))
+	 (apps (or apps eaf-apps-to-install))
+	 (proc
+	  (progn
+	    (async-shell-command (concat
+				  eaf-python-command " install-eaf.py" " --install "
+				  (mapconcat 'symbol-name apps " "))
+				 output-buffer)
+	    (get-buffer-process output-buffer))))
+    (if (process-live-p proc)
+	(set-process-sentinel proc #'eaf--post-install-sentinel)
+      (eaf--post-install))))
+
+(defun eaf--post-install-sentinel (process signal)
+  (when (memq (process-status process) '(exit signal))
```

Comment:
> And the problem with eaf-add-app-dirs-to-load-path not adding the path when run after installing the apps is still unsolved. I am completely clueless there.

Are you sure process-status is `exit` ? 

A process finish, should return `finished` or `exited` status, please check https://www.gnu.org/software/emacs/manual/html_node/elisp/Sentinels.html

Reply:
please reference my another plugin

https://github.com/manateelazycat/multi-term/blob/017c77c550115936860e2ea71b88e585371475d5/multi-term.el#L677

Reply:
Yes, that would also work. Is there an advantage in using that over querying process-status? 

In any case, I am sure that the post-install function is being called.

Reply:
Is post-install call after install-eaf.el finish EAF application git clone?

Reply:
The signal is "finished", so it looks like it. Also, the symlink and byte compilation steps of post-install correctly pick both eaf and the newly installed apps.

Only the load path doesn't get updated.

Reply:
> The signal is "finished", so it looks like it. Also, the symlink and byte compilation steps of post-install both correctly pick eaf and the newly installed apps.
> 
> Only the load path doesn't get updated.

If signal is `finished`, why you use `exit` signal to compare ? 
If signal is mismatch, `eaf--post-install` will call in `eaf--post-install-sentinel `?

Reply:
Signal is the string `"finished"`, the process status is the symbol `exit`. Another accepted value of the process status for the sentinel would be the symbol `signal` (reached if e.g. one were to issue `killall python3` during the installation process).

Comparing a standard list of symbols for possible statuses seems more robust (and efficient) than matching against a non-exhaustive list of signal strings, no?